version: "3"

networks:
  proxy:
    name: proxy

volumes:
  caddy_data:

services:
  caddy:
    image: lucaslorentz/caddy-docker-proxy:2.8.5
    container_name: caddy
    restart: unless-stopped
    networks:
      - proxy
    ports:
      - "8080:80"
    environment:
      - CADDY_INGRESS_NETWORKS=proxy
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - caddy_data:/data

  bitwardenrs:
    image: vaultwarden/server:1.29.1-alpine
    container_name: vaultwarden
    user: 1000:1000
    restart: unless-stopped
    volumes:
      - ${HOMEDIR}/vaultv:/data
    environment:
      WEBSOCKET_ENABLED: "true"
      SENDS_ALLOWED: "true"
      PASSWORD_ITERATIONS: 500000
      SIGNUPS_ALLOWED: "false"
      SIGNUPS_VERIFY: "true"
      DOMAIN: "https://vaultv.${DOMAIN}"
      LOG_FILE: "/data/vaultwarden.log"
      LOG_LEVEL: "warn"
      EXTENDED_LOGGING: "true"
      TZ: "${TIMEZONE}"
    networks:
      - proxy
    labels:
      caddy: http://vaultv.${DOMAIN}
      caddy.reverse_proxy: "{{upstreams 80}}"

  gitea:
    image: gitea/gitea:1.20.2-rootless
    container_name: gitea
    user: 1000:1000
    restart: always
    volumes:
      - ${HOMEDIR}/gitea/data:/var/lib/gitea
      - ${HOMEDIR}/gitea/config:/etc/gitea
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro
    networks:
      - proxy
    labels:
      caddy: http://git.${DOMAIN}
      caddy.reverse_proxy: "{{upstreams 3000}}"
